Source/Core/Core/IOS/DI/DI.cpp

```diff
@@ -30,93 +34,617 @@ void DI::DoState(PointerWrap& p)
 {
   DoStateShared(p);
   p.Do(m_commands_to_execute);
+  p.Do(m_current_partition);
+  p.Do(m_has_initialized);
+  p.Do(m_last_length);
+}
+
+IPCCommandResult DI::Open(const OpenRequest& request)
+{
+  if (!m_has_initialized)
+  {
+    ResetDIRegisters();
+    m_has_initialized = true;
+  }
```

Comment:
This exact snippet appears at least 3 times (from just scrolling through) and should likely be its own function.

Reply:
Done.